packages/nimble-components/src/table-column/text/tests/table-column-text.spec.ts

```diff
@@ -0,0 +1,107 @@
+import { html } from '@microsoft/fast-element';
+import { DesignSystem } from '@microsoft/fast-foundation';
+import type { Table } from '../../../table';
+import { TableColumnText } from '..';
+import { waitForUpdatesAsync } from '../../../testing/async-helpers';
+import { type Fixture, fixture } from '../../../utilities/tests/fixture';
+import type { TableRecord } from '../../../table/types';
+import { TablePageObject } from '../../../table/tests/table.pageobject';
+
+interface SimpleTableRecord extends TableRecord {
+    stringData: string | null;
+    numericData: number | null;
+    booleanData: boolean | null;
+    dateData: Date | null;
+}
+
+const simpleTableData = [
+    {
+        stringData: 'string 1',
+        numericData: 8,
+        booleanData: true,
+        dateData: new Date(2008, 12, 11)
+    },
+    {
+        stringData: '<button></button>',
+        numericData: 0,
+        booleanData: true,
+        dateData: new Date(2022, 5, 30)
+    },
+    {
+        stringData: null,
+        numericData: null,
+        booleanData: null,
+        dateData: null
+    }
+] as const;
+
+async function setup(): Promise<Fixture<Table<SimpleTableRecord>>> {
+    return fixture<Table<SimpleTableRecord>>(
+        html`<nimble-table>
+                <${DesignSystem.tagFor(
+        TableColumnText
+    )} field-name="stringData" placeholder="no value">stringData</${DesignSystem.tagFor(
+    TableColumnText
+)}>
+                <${DesignSystem.tagFor(
+        TableColumnText
+    )} field-name="numericData">numericData</${DesignSystem.tagFor(
+    TableColumnText
+)}>
+                <${DesignSystem.tagFor(
+        TableColumnText
+    )} field-name="booleanData">booleanData</${DesignSystem.tagFor(
+    TableColumnText
+)}>
+                <${DesignSystem.tagFor(
+        TableColumnText
+    )} field-name="dateData">dateData</${DesignSystem.tagFor(
+    TableColumnText
+)}>
+                <button>Ignore Me</button>
+            </nimble-table>`
+    );
+}
+
+describe('TableColumnText', () => {
+    let element: Table<SimpleTableRecord>;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+    let pageObject: TablePageObject<SimpleTableRecord>;
+
+    beforeEach(async () => {
+        ({ element, connect, disconnect } = await setup());
+        pageObject = new TablePageObject<SimpleTableRecord>(element);
+    });
+
+    afterEach(async () => {
+        await disconnect();
+    });
+
+    it('displays placeholder string when no value present', async () => {
```

Comment:
I think we should add tests for some more corner cases:
1. `null` data value should be rendered as the placeholder text
2. The string `'null'` should render as that string, not the placeholder (not normally a test case I'd suggest, but our current table gets this wrong)
3. Should empty string should render as placeholder or empty string?
4. non-ASCII strings
5. other interesting strings from [BLNS](https://github.com/minimaxir/big-list-of-naughty-strings/blob/master/blns.txt)
6. Tests for various values of `placeholder` (what if it's null/undefined/HTML content/non-ASCII)?
7. long strings overflow behavior (Might be a Chromatic test or could wait until we figure out tooltips.)
8. poorly implemented column types which don't specify valid data fields or which throw in their implementation. We should ensure these generate good exceptions / invalid states

Reply:
I've addressed a subset of the cases you outlined. Namely 1, 2, 3, and parts of 6 and 8. My recommendation is to visit these other cases after the initial PR goes in (perhaps immediately).

Reply:
Thanks for the new tests. I'm ok with deferring the additional tests to unblock other work as long as you track it in an issue (either a new one or a task in an existing one) and tackle them in a quick followup.

Reply:
Yep, I intend to try and address these other test cases in an immediate followup.